### PR TITLE
Feat/tokens + darkmode + theme support

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?
Well ALOT.

This PR tokenises everything and also introduces dark mode across the site. 

YEP its massive! However, a lot of the changes are simply swapping one or two classnames per file. 

I'd recommend the figma file for a breakdown of colors

https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS?node-id=25883-174646

The tailwind preset also contains names of classes that we will be using for the forceable feature.

If anyone here fancies creating a console PR for this i'd love you a lot - all that really needs to happen to ensure components work - is we copy over the styles from global.css for `:root` - We do not plan to use darkmode on console.  

Only some pages have lightmode forced. This is to present a more business esk approach for onboarding etc etc 

This is too big to loom sorry - Please pull down or explore preview URL 

Screenshot 2023-04-03 at 11 14 54

Website: https://github.com/calcom/website/pull/467
Console: https://github.com/calcom/console/pull/93

### CODE REVIEW

File: packages/lib/getBrandColours.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      "booking-highlight-color": embedBrandingColors.highlightColor || "#10B981",
+      "booking-lightest-color": embedBrandingColors.lightestColor || "#E1E1E1",
+      "booking-lighter-color": embedBrandingColors.lighterColor || "#ACACAC",
+      "booking-light-color": embedBrandingColors.lightColor || "#888888",
+      "booking-median-color": embedBrandingColors.medianColor || "#494949",
+      "booking-dark-color": embedBrandingColors.darkColor || "#313131",
+      "booking-darker-color": embedBrandingColors.darkerColor || "#292929",
+    },
+  };
+  return theme;
```
Comment (sean-brydon) [inline packages/lib/getBrandColours.tsx:190]: @hariombalhara can we move these to normal tokens we use ? 

`bg-emphasis` `bg-subtle` etc? If not this PR might break embed styling - i tested and it seemed OKAY

File: packages/lib/getBrandColours.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+  // convert to 6 digit equivalent if 3 digit code is entered
+  lightVal = normalizeHexCode(lightVal, false);
+
+  darkVal = normalizeHexCode(darkVal, true);
+
+  const lightColourMap = createColorMap(lightVal);
+  const darkColourMap = createColorMap(darkVal);
+
+  const theme = {
```
Comment (sean-brydon) [inline packages/lib/getBrandColours.tsx:166]: Return a theme object that we use in 

`useCalcomTheme` to set these when brand colors change. 

This approach i believe was easier and nicer to use than the custom branding file. Had a discussion with @keithwillcode about it - happy to change back if anyone disagrees

File: apps/storybook/components/VariantsTable.tsx [hunk 3]
```diff
@@ -42,7 +43,7 @@ export function VariantsTable({
         </table>
       </div>
       {!isDark && (
-        <div data-mode="dark">
+        <div data-mode="dark" className="dark">
```
Comment (sean-brydon) [inline apps/storybook/components/VariantsTable.tsx:46]: Force darkmode in storybook now

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 4]
```diff
@@ -58,12 +58,24 @@ const dateQuerySchema = z.object({
...
 const AvailabilityPage = ({ profile, eventType, ...restProps }: Props) => {
   const router = useRouter();
   const isEmbed = useIsEmbed(restProps.isEmbed);
   const query = dateQuerySchema.parse(router.query);
   const { rescheduleUid } = query;
   useTheme(profile.theme);
+  useBrandColors({
+    brandColor: profile.brandColor,
+    darkBrandColor: profile.darkBrandColor,
+  });
```
Comment (sean-brydon) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:78]: New way of handling brand colours

File: apps/web/pages/settings/my-account/calendars.tsx [hunk 5]
```diff
@@ -84,17 +84,17 @@ const CalendarsView = () => {
         success={({ data }) => {
           return data.connectedCalendars.length ? (
             <div>
-              <div className="mt-4 flex space-x-4 rounded-md border-gray-200 bg-gray-50 p-2 sm:mx-0 sm:p-10 md:border md:p-6 xl:mt-0">
-                <div className=" flex h-9 w-9 items-center justify-center rounded-md border-2 border-gray-200 bg-white p-[6px]">
+              <div className="bg-muted mt-4 flex space-x-4 rounded-md border-subtle p-2 sm:mx-0 sm:p-10 md:border md:p-6 xl:mt-0">
+                <div className=" bg-default flex h-9 w-9 items-center justify-center rounded-md border-2 border-subtle p-[6px]">
```
Comment (Udit-takkar) [inline apps/web/pages/settings/my-account/calendars.tsx:88]: Screenshot 2023-04-03 at 4 46 47 PM

File: apps/web/pages/settings/my-account/appearance.tsx [hunk 6]
```diff
@@ -93,8 +93,8 @@ const AppearanceView = () => {
       <Meta title={t("appearance")} description={t("appearance_description")} />
       <div className="mb-6 flex items-center text-sm">
         <div>
-          <p className="font-semibold">{t("theme")}</p>
-          <p className="text-gray-600">{t("theme_applies_note")}</p>
+          <p className="text-default font-semibold">{t("theme")}</p>
+          <p className="text-default">{t("theme_applies_note")}</p>
```
Comment (Udit-takkar) [inline apps/web/pages/settings/my-account/appearance.tsx:97]: Screenshot 2023-04-03 at 4 48 10 PM

Comment (sean-brydon) [reply apps/web/pages/settings/my-account/appearance.tsx:97]: Added to my list

**Comment by CarinaWolli** [comment]
Some things I found when clicking through workflows:

- Phone number is in black and not visible
Screenshot 2023-04-03 at 13 00 23

- Skeleton loader for workflow list has white lines (doesn’t have that in event-types)
Screenshot 2023-04-03 at 13 01 06

- Editor: wrong color on hover + to much space on the right after the dropdown
Screenshot 2023-04-03 at 13 04 26

- Background white on hover in workflow list
Screenshot 2023-04-03 at 13 17 39

- Event-types: space between icon and text missing
Screenshot 2023-04-03 at 12 54 54

File: apps/web/pages/settings/my-account/appearance.tsx (hunk 6 above)
Comment (Udit-takkar) [reply apps/web/pages/settings/my-account/appearance.tsx:97]: this doesn't look right

Comment (sean-brydon) [reply apps/web/pages/settings/my-account/appearance.tsx:97]: Will fix <3

**Review by Udit-takkar — DISMISSED** [review]
- [x]  switches track color does not change here 

https://user-images.githubusercontent.com/53316345/229495443-f4b8972c-9b70-4dd9-a8d3-eaf4b1f91ac7.mov

- [x] there was was progress bars of steps when creating team

Screenshot 2023-04-03 at 4 54 54 PM

**Comment by Udit-takkar** [comment]
Screenshot 2023-04-03 at 5 03 58 PM

- not enough margin between them

Screenshot 2023-04-03 at 5 04 37 PM

- dropdown items 

Screenshot 2023-04-03 at 5 06 00 PM

- 
Screenshot 2023-04-03 at 5 08 05 PM

- delete icon here 

Screenshot 2023-04-03 at 5 08 23 PM

- text not visible in date overrdies

Screenshot 2023-04-03 at 5 10 04 PM

- text color in dropdown

Screenshot 2023-04-03 at 5 13 30 PM

- no avatar text

Screenshot 2023-04-03 at 5 14 27 PM

- switch track color not changing on toggling

Screenshot 2023-04-03 at 5 15 42 PM

- some icons of apps are not visible in dark mode

Screenshot 2023-04-03 at 5 18 09 PM

- after new app install

Screenshot 2023-04-03 at 5 20 19 PM

**Review by PeerRich — APPROVED** [review]
terrific work! lets follow up with smaller PRs for any leftovers

File: apps/web/components/team/screens/Team.tsx [hunk 7]
```diff
@@ -18,20 +19,20 @@ const Member = ({ member, teamName }: { member: MemberType; teamName: string | n
...
         <section className="line-clamp-4 mt-2 w-full space-y-1">
-          <p className="font-medium text-gray-900 dark:text-white">{member.name}</p>
-          <div className="line-clamp-3 overflow-ellipsis text-sm font-normal text-gray-500 dark:text-white">
+          <p className="text-emphasis dark:text-inverted font-medium">{member.name}</p>
+          <div className="line-clamp-3 text-subtle dark:text-inverted overflow-ellipsis text-sm font-normal">
             {!isBioEmpty ? (
               <>
                 <div
-                  className="dark:text-darkgray-600 text-sm text-gray-500 [&_a]:text-blue-500 [&_a]:underline [&_a]:hover:text-blue-600"
-                  dangerouslySetInnerHTML={{ __html: member.safeBio || "" }}
```
Comment (zomars) [inline apps/web/components/team/screens/Team.tsx:34]: Why was this removed?

File: apps/web/pages/404.tsx [hunk 8]
```diff
@@ -215,147 +215,151 @@ export default function Custom404() {
             </div>
           ) : (
             <>
```
Comment (zomars) [inline apps/web/pages/404.tsx:217]: Why the double fragment?

```suggestion
```

File: apps/web/pages/[user].tsx [hunk 9]
```diff
@@ -118,53 +118,39 @@ export default function User(props: inferSSRProps<typeof getServerSideProps> & E
...
-                {user.verified && (
-                  <BadgeCheckIcon className="mx-1 -mt-1 inline h-6 w-6 text-blue-500 dark:text-white" />
-                )}
+                {user.verified && <BadgeCheckIcon className=" mx-1 -mt-1 inline h-6 w-6 text-blue-500" />}
               </h1>
               {!isBioEmpty && (
                 <>
                   <div
-                    className=" dark:text-darkgray-600 text-sm text-gray-500 [&_a]:text-blue-500 [&_a]:underline [&_a]:hover:text-blue-600"
-                    dangerouslySetInnerHTML={{ __html: props.safeBio }}
```
Comment (zomars) [inline apps/web/pages/[user].tsx:150]: Also this is removing safeBio as well

File: apps/web/pages/[user].tsx [hunk 10]
```diff
@@ -204,14 +188,14 @@ export default function User(props: inferSSRProps<typeof getServerSideProps> & E
...
           </div>
+
           {isEventListEmpty && <EmptyPage name={user.name ?? "User"} />}
         </main>
         <Toaster position="bottom-right" />
       </div>
     </>
   );
 }
-User.isThemeSupported = true;
```
Comment (zomars) [inline apps/web/pages/[user].tsx:214]: Is this unsupported?

File: apps/web/pages/[user]/[type].tsx [hunk 11]
```diff
@@ -54,8 +54,6 @@ export default function Type(props: AvailabilityPageProps) {
   );
 }
 
-Type.isThemeSupported = true;
```
Comment (zomars) [inline apps/web/pages/[user]/[type].tsx:57]: Intended?

File: apps/web/pages/[user].tsx (hunk 10 above)
Comment (sean-brydon) [reply apps/web/pages/[user].tsx:214]: User.isThemeSupported = true 
and 
undefined 
means we enable dark/light mode

Component.isThemeSupported = false -> we force light mode

File: apps/web/pages/[user]/[type].tsx (hunk 11 above)
Comment (sean-brydon) [reply apps/web/pages/[user]/[type].tsx:57]: Yes 

https://github.com/calcom/cal.com/pull/7876#discussion_r1157719251 :)

File: packages/app-store/zapier/components/TemplateCard.tsx [hunk 12]
```diff
@@ -4,7 +4,7 @@ import type { Template } from "./AppSettingsInterface";
 
 export default function TemplateCard({ template }: { template: Template }) {
   return (
-    <div className="min-h-16 sticky flex flex-col justify-between gap-4 rounded-md border border-gray-200 bg-white p-5">
+    <div className="min-h-16 bg-default sticky flex flex-col justify-between gap-4 rounded-md border border-subtle p-5">
```
Comment (zomars) [inline packages/app-store/zapier/components/TemplateCard.tsx:7]: Linting error here

File: packages/features/ee/support/components/HelpMenuItem.tsx [hunk 13]
```diff
@@ -209,8 +209,8 @@ export default function HelpMenuItem({ onHelpItemSelect }: HelpMenuItemProps) {
         <span> {t("or").toLowerCase()} </span>
         <a
           onClick={() => onHelpItemSelect()}
-          className="font-medium underline hover:text-gray-700"
-          href="https://cal.com/docs/"
+          className="hover:text-emphasis text-defualt font-medium underline"
+          href="https://docs.cal.com/"
```
Comment (zomars) [inline packages/features/ee/support/components/HelpMenuItem.tsx:213]: Faulty merge here:

```suggestion
          href="https://cal.com/docs/"
```

Comment (zomars) [reply packages/features/ee/support/components/HelpMenuItem.tsx:213]: The permalink changed in #7782

**Review by zomars — DISMISSED** [review]
Blocking to address some merge issues from main

File: packages/features/eventtypes/components/EventTypeDescription.tsx [hunk 14]
```diff
@@ -49,11 +49,11 @@ export const EventTypeDescription = ({
 
   return (
     <>
-      <div className={classNames("dark:text-darkgray-800 text-gray-500", className)}>
+      <div className={classNames(" text-subtle", className)}>
```
Comment (zomars) [inline packages/features/eventtypes/components/EventTypeDescription.tsx:52]:
```suggestion
      <div className={classNames("text-subtle", className)}>
```

File: packages/features/eventtypes/components/EventTypeDescription.tsx [hunk 15]
```diff
@@ -49,11 +49,11 @@ export const EventTypeDescription = ({
...
         {eventType.description && (
           <div
             className={classNames(
-              "dark:text-darkgray-800 max-w-[280px] break-words py-1 text-sm text-gray-500 sm:max-w-[500px] [&_a]:text-blue-500 [&_a]:underline [&_a]:hover:text-blue-600",
+              " text-subtle max-w-[280px] break-words py-1 text-sm sm:max-w-[500px] [&_a]:text-blue-500 [&_a]:underline [&_a]:hover:text-blue-600",
```
Comment (zomars) [inline packages/features/eventtypes/components/EventTypeDescription.tsx:56]:
```suggestion
              "text-subtle max-w-[280px] break-words py-1 text-sm sm:max-w-[500px] [&_a]:text-blue-500 [&_a]:underline [&_a]:hover:text-blue-600",
```

**Comment by PeerRich** [comment]
still adding some last minute improvements

File: packages/lib/CustomBranding.test.tsx
Comment (zomars) [inline packages/lib/CustomBranding.test.tsx:1]: Are these tests irrelevant now?

File: packages/lib/getBrandColours.test.ts
Comment (zomars) [inline packages/lib/getBrandColours.test.ts:1]: Just answered my own question

File: packages/ui/components/apps/AppCard.tsx [hunk 16]
```diff
@@ -39,14 +40,22 @@ export function AppCard({ app, credentials, searchText }: AppCardProps) {
   useEffect(() => {
     setSearchTextIndex(searchText ? app.name.toLowerCase().indexOf(searchText.toLowerCase()) : undefined);
   }, [app.name, searchText]);
-
+  console.log({ app });
```
Comment (zomars) [inline packages/ui/components/apps/AppCard.tsx:43]:
```suggestion
```

File: packages/ui/components/button/Button.tsx [hunk 17]
```diff
@@ -62,80 +62,77 @@ const buttonClasses = cva(
         true: "cursor-not-allowed",
       },
     },
+    // FIXME
+    // TODO: come back to this and fix before merging. Buttons are all over the shop some use brand colour some use gray.
```
Comment (zomars) [inline packages/ui/components/button/Button.tsx:66]: Is this still relevant?

File: packages/ui/components/card/Card.tsx [hunk 18]
```diff
@@ -148,12 +148,13 @@ export function Card({
               onClick={actionButton?.onClick}
               target="_blank"
               rel="noreferrer"
-              className="text-xs font-medium">
+              prefetch={!learnMore.href.startsWith("https://cal.com/blog")}
```
Comment (zomars) [inline packages/ui/components/card/Card.tsx:151]: Is this prefect coming from this PR or is it reintroducing previously removed code?

File: packages/ui/components/editor/plugins/ToolbarPlugin.tsx [hunk 19]
```diff
@@ -426,42 +426,53 @@ export default function ToolbarPlugin(props: TextEditorProps) {
...
+              className={isBold ? "bg-subtle" : ""}
+            />
+            // <button
+            //   type="button"
+            //   onClick={() => {
+
+            //   }}
+            //   className={"toolbar-item spaced rounded-md " + (isBold ? "active" : "")}
+            //   aria-label="Format Bold">
+            // </button>
```
Comment (zomars) [inline packages/ui/components/editor/plugins/ToolbarPlugin.tsx:449]: Removed if unused

File: packages/app-store/zapier/components/TemplateCard.tsx (hunk 12 above)
Comment (zomars) [inline packages/app-store/zapier/components/TemplateCard.tsx:7]:
```suggestion
    <div className="min-h-16 bg-default border-subtle sticky flex flex-col justify-between gap-4 rounded-md border p-5">
```

File: packages/ui/components/card/Card.tsx (hunk 18 above)
Comment (hariombalhara) [reply packages/ui/components/card/Card.tsx:151]: I think it got removed in a recent PR and being accidentally added here.

**Comment by Udit-takkar** [comment]
just this page is left  @sean-brydon 

image

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Comment by hariombalhara** [comment]
- Routing Forms listing screen when upgrade is required Screenshot 2023-04-05 at 1 20 36 PM
- Loader not clearly visible on button Screenshot 2023-04-05 at 1 23 01 PM

File: apps/web/components/team/screens/Team.tsx [hunk 20]
```diff
@@ -2,6 +2,7 @@ import Link from "next/link";
 
 import { WEBAPP_URL } from "@calcom/lib/constants";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { md } from "@calcom/lib/markdownIt";
```
[the remark this answers is not in the document]
Comment (JeroenReumkens) [reply apps/web/components/team/screens/Team.tsx:5]: I think it would make sense to do this on the server side, as a prisma middleware? We already do this for the event types as well (`packages/prisma/middleware/eventTypeDescriptionParseAndSanitize.ts`). Although I did see that sometimes we tend to do this in other places as well. Keeping it in the prisma middleware makes the most sense I think.

File: packages/lib/getBrandColours.tsx (hunk 1 above)
Comment (JeroenReumkens) [reply packages/lib/getBrandColours.tsx:204]: In my new booker I haven't implemented them yet, because I was waiting for these new variables to become available. I think it makes much more sense to replace them with the new variables, instead of keeping variables specific for only the booker. That way if users use these variables, and will use more of our atoms later, they all share the same variables and thus are styled right away.

File: apps/web/styles/globals.css [hunk 21]
```diff
@@ -2,19 +2,99 @@
 @tailwind components;
 @tailwind utilities;
 
-
+:root {
```
Comment (JeroenReumkens) [inline apps/web/styles/globals.css:5]: Shall we add `color-scheme: dark;` here? That makes the scrollbars default dark instead of light as well, looks a lot better I think. 

See for example the text area scrollbars:

before:
CleanShot 2023-04-05 at 11 51 53@2x

after:
CleanShot 2023-04-05 at 11 52 22@2x


Both cases are ugly because of cut offs etc, but I think the automatic darker scrollbar already improves it by a lot.

Comment (JeroenReumkens) [reply apps/web/styles/globals.css:5]: Pushed the change.

**Review by JeroenReumkens — APPROVED** [review]
Amazing work sean! 🔥

**Review by hariombalhara — APPROVED** [review]
Tested. Looks good to me. Beyond amazing work @sean 🙏
